### ISSUE TITLE
How to output multiple PSMs per spectrum?

### ISSUE THREAD

**Issue by glormph** [percolator/percolator#360]
Hi, I'm using MSGFPlus in the "only output rank 1 PSMs" mode, and then `msgf2pin` with `-m 4` to also extract all the solutions that have an identical score in MSGF:
- isoleucine/leucine
- oxidation on different nearby M
- different short sequence somehow (I hardly believed it, but it happens!)

Anyway, the percolator input file contains all of these (with correctly different PSM ids for each), but when running percolator I only find one solution per scan in the output. I found some issues that mention doing multi-solution analysis but they weren't mentioning this behavior, making me think I'm doing something wrong. Maybe I just couldn't find the corresponding option for this for the percolator process?

Command lines are:
```
msgf2pin -m 4 -o percoin.tsv -e trypsin -P "decoy_" metafile
percolator -Y -j percoin.tsv -X perco.xml -N 500000 --decoy-xml-output
```

**Comment by glormph** [comment]
I've poked around a bit more, and it only happens using `-Y` for target/decoy competition (I'm searching using a concatenated DB with multiple files so I use `-Y` to not use mixmax due to duplicate scan numbers), so I'm wondering if it is intentional. On the whole, of a dataset of 22 PSMs, 13 were retained in my tests. Indeed, there are only 13 scan numbers in the input, but multiple sequences and 22 different psm_ids, which differ in their SIR (`SpectrumIdentificationResult` id?). When not using `-Y` all of them are in the XML output.

**Comment by MatthewThe** [comment]
Yes, `--search-input concatenated` uses the FDR method for concatenated searches. We specifically introduced this flag to deal with non-standard scenarios such as yours.

**Comment by glormph** [comment]
Too many options not to get confused :)
So, if I have understood correctly, I used to do `-Y` on my concatenated search, which avoids mixmax, but is actually meant for when one runs separated searches and do the competition step inside percolator? So when I instead run `-I concatenated` it will use the same FDR (non mixmax) but not do actual competition based on PSM scan numbers.
And when one runs `msgf2pin` without `-m`, the behaviour of `-Y` and `-I concatenated` would be more or less identical, since it does not have multiple solutions for a given scan numbers.

Hope I have understood it now, thanks for the quick help! :)
